### PR TITLE
Fixing issues on rounded corners + borders

### PR DESCRIPTION
**Pull request by viteinfinite**
Allow borders to be correctly drawn on views having rounded corners.
Also, fixing a major visual issue on antialiasing which prevented borders from having smooth corners.

### CODE REVIEW

File: Gaikan/UIKit/Layer/BorderLayer.swift [hunk 1]
```diff
@@ -27,19 +46,19 @@ class BorderLayer : CALayer {
             case .top:
                 self.draw(border: border,
                           path: path,
-                          points: [ CGPoint(x: 0, y: 0), CGPoint(x: ctx.width, y: 0), CGPoint(x: ctx.width, y: Int(border.width)), CGPoint(x: 0, y: Int(border.width)) ])
+                          points: [ CGPoint(x: 0, y: 0), CGPoint(x: bounds.width, y: 0), CGPoint(x: bounds.width, y: border.width), CGPoint(x: 0, y: border.width) ])
```
Comment (pjechris) [inline Gaikan/UIKit/Layer/BorderLayer.swift:49]: I guess this is correct but IMO `ctx.width` feels more correct as we can't be 100% sure that ctx == bounds

Comment (viteinfinite) [reply Gaikan/UIKit/Layer/BorderLayer.swift:49]: Thing is that ctx depends on screen scale. So, if `layer.contentsScale` is 2.0 (for instance), `ctx.width` would be twice the bound width.

Comment (pjechris) [reply Gaikan/UIKit/Layer/BorderLayer.swift:49]: Oh ok, didn't know that!

**Comment by viteinfinite** [comment]
🎉
Thanks!

**Pull request merged**